src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -81,6 +81,7 @@ void GcsNodeManager::HandleGetClusterId(rpc::GetClusterIdRequest request,
 void GcsNodeManager::HandleRegisterNode(rpc::RegisterNodeRequest request,
                                         rpc::RegisterNodeReply *reply,
                                         rpc::SendReplyCallback send_reply_callback) {
+  // This function invokes a read lock
```

Comment:
but the code below doesn't hold a read lock until the on_done lambda (which takes an exclusive lock). Let's update the comment.

Reply:
That's not quite right, we grab a scoped read lock a bit further down at line 109 in the main body of the function while iterating over the node list.